### PR TITLE
Add panic version of common operations

### PR DESCRIPTION
**Pull request by dascapytal1559**
I've found these Musts to be very useful in my projects so I thought I'd pitch it. 

Feel free to object.

### CODE REVIEW

**Comment by pmzajaczkowski** [comment]
@govalues Could you please merge it? :)

**Comment by eapenkin** [comment]
Dear @dascapytal1559,

Thank you for your suggestion! I understand how "Must" methods can be convenient for simplifying code by avoiding explicit error handling. However, the primary goal of this package is to prioritize safety and robustness in decimal computations.

That said, I’d like to share a couple of suggestions that might help achieve similar convenience while keeping computations clear and panic-free:

**Function per Formula**: Consider breaking down complex formulas into smaller functions. Each function can represent a meaningful part of the computation. This approach not only improves readability but also enhances maintainability by giving clear names to specific steps in the computation.

**Favor Composite Methods**: Where possible, simplify your formulas algebraically to make use of the composite operations available in the library (e.g., `Sub`, `Prod`, `AddMul`, `AddSub`, `SubMul`, `SubQuo`). These methods are optimized for performance, accuracy, and readability.

I hope these suggestions are helpful!

**Comment by dascapytal1559** [comment]
I accept panic-free is a core tenent of this package but I also dont find the suggestions helpful in resolving my pain-point using this package. Thanks for considering anyway, I'll continue to run a fork for my usecases. Feel free to close the PR.

File: decimal.go [hunk 1]
```diff
@@ -709,6 +709,29 @@ func (d Decimal) MarshalText() ([]byte, error) {
 	return []byte(d.String()), nil
 }
 
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (d *Decimal) UnmarshalJSON(s []byte) error {
+	var err error
+	unquoted := string(s[1 : len(s)-1])
+	*d, err = Parse(unquoted)
+	return err
+}
```
Comment (ccoVeille) [inline decimal.go:718]: Here you could call Unquote


```suggestion
// UnmarshalJSON implements the json.Unmarshaler interface.
func (d *Decimal) UnmarshalJSON(s []byte) error {
	var err error
	unquoted := strings.Unquote(string(s))
	*d, err = Parse(unquoted)
	return err
}
```

But also, you could simply do this


```suggestion
// UnmarshalText implements the [json.UnmarshalerText] interface.
func (d *Decimal) UnmarshalText(s []byte) error {
	var err error
	*d, err = Parse(string(s))
	return err
}
```

File: decimal.go [hunk 2]
```diff
@@ -709,6 +709,29 @@ func (d Decimal) MarshalText() ([]byte, error) {
 	return []byte(d.String()), nil
 }
 
+// UnmarshalJSON implements the json.Unmarshaler interface.
```
Comment (ccoVeille) [inline decimal.go:712]:
```suggestion
// UnmarshalJSON implements the [json.Unmarshaler] interface.
```

File: decimal.go [hunk 3]
```diff
@@ -709,6 +709,29 @@ func (d Decimal) MarshalText() ([]byte, error) {
...
+	var err error
+	unquoted := string(s[1 : len(s)-1])
+	*d, err = Parse(unquoted)
+	return err
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+func (d Decimal) MarshalJSON() ([]byte, error) {
+	return []byte("\"" + d.String() + "\""), nil
+}
```
Comment (ccoVeille) [inline decimal.go:723]:
```suggestion
// MarshalText implements the [json.MarshalerText] interface.
func (d Decimal) MarshalText() ([]byte, error) {
	return []byte("\"" + d.String() + "\""), nil
}
```

File: decimal.go [hunk 4]
```diff
@@ -709,6 +709,29 @@ func (d Decimal) MarshalText() ([]byte, error) {
...
+
+// GobDecode implements the gob.GobDecoder interface for gob serialization.
+func (d *Decimal) GobDecode(data []byte) error {
+	return d.UnmarshalBinary(data)
+}
+
+// GobEncode implements the gob.GobEncoder interface for gob serialization.
+func (d Decimal) GobEncode() ([]byte, error) {
+	return d.MarshalBinary()
+}
```
Comment (ccoVeille) [inline decimal.go:733]:
```suggestion
// GobDecode implements the [gob.GobDecoder] interface for gob serialization.
func (d *Decimal) GobDecode(data []byte) error {
	return d.UnmarshalBinary(data)
}

// GobEncode implements the [gob.GobEncoder] interface for gob serialization.
func (d Decimal) GobEncode() ([]byte, error) {
	return d.MarshalBinary()
}
```

File: decimal_test.go [hunk 5]
```diff
@@ -70,6 +88,198 @@ func TestDecimal_Interfaces(t *testing.T) {
 	}
 }
 
+func TestDecimal_UnmarshallJson(t *testing.T) {
```
Comment (ccoVeille) [inline decimal_test.go:91]:
```suggestion
func TestDecimal_UnmarshallJSON(t *testing.T) {
```

File: decimal_test.go [hunk 6]
```diff
@@ -70,6 +88,198 @@ func TestDecimal_Interfaces(t *testing.T) {
...
+			}
+
+			if data.Value.Cmp(tt.expected) != 0 {
+				t.Errorf("json.Unmarshal() failed, got=%q, want=%q", data.Value, tt.expected)
+			}
+		})
+	}
+}
+
+func TestDecimal_MarshallJson(t *testing.T) {
```
Comment (ccoVeille) [inline decimal_test.go:148]:
```suggestion
func TestDecimal_MarshallJSON(t *testing.T) {
```

**Pull request closed without merging**